src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -294,11 +409,55 @@ public DetailedPackageMetadata PackageMetadata
                 if (_packageMetadata != value)
                 {
                     _packageMetadata = value;
+
+                    if (_packageMetadata?.DeprecationMetadata != null)
+                    {
+                        PackageDeprecationReasonsExplained = ExplainPackageDeprecationReasons(_packageMetadata.DeprecationMetadata.Reasons);
+                        PackageDeprecationReasons = "(" + string.Join(", ", _packageMetadata.DeprecationMetadata.Reasons) + ")";
+                        PackageDeprecationCustomMessage = string.IsNullOrWhiteSpace(_packageMetadata.DeprecationMetadata.Message) ? null : _packageMetadata.DeprecationMetadata.Message;
+
+                        var alternatePackage = _packageMetadata.DeprecationMetadata.AlternatePackage;
+                        if (alternatePackage != null)
+                        {
+                            PackageDeprecationAlternatePackageLinkText = GetPackageDeprecationAlternatePackageLinkText(alternatePackage);
+
+                            PackageDeprecationAlternatePackageLinkUrl = new Uri(
+                                    _packageMetadata?.PackageDetailsUrl
```

Comment:
Package details URL is only implemented by nuget.org AFAIK. We should not depend on this.

I think using the package details URL here is wrong. We have a way to view package details in the UI already -- that's the side pane that this is showing up in.

Even if we did want to go this way we would use `PackageDetailsUriResourceV3Provider` not string replacement.